pkgs/by-name/cm/cmake/package.nix

```diff
@@ -101,7 +101,7 @@ stdenv.mkDerivation (finalAttrs: {
   ++ lib.optional stdenv.isDarwin CoreServices
   ++ lib.optional (stdenv.isDarwin && !isMinimalBuild) SystemConfiguration;
 
-  propagatedBuildInputs = lib.optional stdenv.isDarwin ps;
+  propagatedBuildInputs = lib.optional (stdenv.isDarwin || stdenv.isFreeBSD) ps;
```

Comment:
Why not propagate it over Linux too?

Reply:
Propagation is weird and difficult to debug and should be avoided where possible.

What's this used for?  The library or the executable?  If executable, could we wrap instead?

Reply:
Looks to be both. However, the place it's used is extremely well isolated and unlikely to change (it's a vendored lib), so we can assuredly just patch it to use an absolute path. This can remove the propagation on Darwin as well.

I suspect ps is part of stdenv on linux so it is not needed there, so I will make the patch only apply to Darwin | FreeBSD in order to minimize rebuilds.